src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1029,6 +1030,14 @@ def _validate_vm_nic_delete_option(namespace):
                                             "specify delete option for the new nic created: --nic-delete-option Delete")
 
 
+def _validate_vm_data_disk_delete_option(namespace):
+    if "attach_data_disks" not in namespace:
+        return
+    if not namespace.attach_data_disks and namespace.data_disk_delete_option:
+        if len(namespace.data_disk_delete_option) == 1 and len(namespace.data_disk_delete_option[0].split('=')) == 1:  # pylint: disable=line-too-long
+            namespace.data_disk_delete_option = namespace.data_disk_delete_option[0]
```

Comment:
>   if attach_data_disks and data_disk_delete_option:
      data_disk_delete_option = validate_delete_options(attach_data_disks, data_disk_delete_option)

When `amespace.attach_data_disks` is `None`, the above code will not be executed, is the logic here useful ?